From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: twisti <76837088+twisti-dev@users.noreply.github.com>
Date: Fri, 8 May 2026 17:03:14 +0200
Subject: [PATCH] Rebrand to SurfCanvas


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 4e9e4e40d31a323bf31aa3cb08ba10f94ebc053c..683274ecaf75d9ad8f525d87aeba064eb24e5796 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -26,6 +26,12 @@ public interface ServerBuildInfo {
      */
     Key BRAND_CANVAS_ID = Key.key("canvasmc", "canvas");
     // Canvas end - Rebrand
+    // SurfCanvas start - Rebrand
+    /**
+     * The brand id for SurfCanvas
+     */
+    Key BRAND_SURF_CANVAS_ID = Key.key("slne", "surfcanvas");
+    // SurfCanvas end - Rebrand
 
     /**
      * Gets the {@code ServerBuildInfo}.
