From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: twisti <76837088+twisti-dev@users.noreply.github.com>
Date: Fri, 8 May 2026 17:03:14 +0200
Subject: [PATCH] Rebrand to SurfCanvas


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 2d65442da4127f9eff2fa9a769e19517f87ea101..49e9aa5508f41bc24906b1edcfcb10d24f3d87bd 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -28,6 +28,12 @@ public interface ServerBuildInfo {
      */
     Key BRAND_CANVAS_ID = Key.key("canvasmc", "canvas");
     // Canvas end - Rebrand
+    // SurfCanvas start - Rebrand
+    /**
+     * The brand id for SurfCanvas
+     */
+    Key BRAND_SURF_CANVAS_ID = Key.key("slne", "surfcanvas");
+    // SurfCanvas end - Rebrand
 
     /**
      * Gets the {@code ServerBuildInfo}.
