--- a/src/main/java/io/canvasmc/canvas/Config.java
+++ b/src/main/java/io/canvasmc/canvas/Config.java
@@ -553,6 +_,8 @@
             " - purpur.enderchest.rows.one"
         })
         public boolean enderChestPermissionRows = false;
+        @Comment("Whether to persist hidden ender chest rows")
+        public boolean enderChestPersistHiddenRows = true;
     }
 
     @Comment("Disables leaf decaying")
