--- a/net/minecraft/world/inventory/PlayerEnderChestContainer.java
+++ b/net/minecraft/world/inventory/PlayerEnderChestContainer.java
@@ -47,19 +_,25 @@
     }
 
     public void fromSlots(ValueInput.TypedInputList<ItemStackWithSlot> input) {
-        for (int i = 0; i < this.getContainerSize(); i++) {
+        // Purpur start - Barrels and enderchests 6 rows
+        int storageSlotCount = io.canvasmc.canvas.Config.INSTANCE.containers.enderChestSixRows && io.canvasmc.canvas.Config.INSTANCE.containers.enderChestPersistHiddenRows ? 54 : this.getContainerSize();
+        for (int i = 0; i < storageSlotCount; i++) {
+            // Purpur end - Barrels and enderchests 6 rows
             this.setItem(i, ItemStack.EMPTY);
         }
 
         for (ItemStackWithSlot itemStackWithSlot : input) {
-            if (itemStackWithSlot.isValidInContainer(this.getContainerSize())) {
+            if (itemStackWithSlot.isValidInContainer(storageSlotCount)) { // Purpur - Barrels and enderchests 6 rows
                 this.setItem(itemStackWithSlot.slot(), itemStackWithSlot.stack());
             }
         }
     }
 
     public void storeAsSlots(ValueOutput.TypedOutputList<ItemStackWithSlot> output) {
-        for (int i = 0; i < this.getContainerSize(); i++) {
+        // Purpur start - Barrels and enderchests 6 rows
+        int storageSlotCount = io.canvasmc.canvas.Config.INSTANCE.containers.enderChestSixRows && io.canvasmc.canvas.Config.INSTANCE.containers.enderChestPersistHiddenRows ? 54 : this.getContainerSize(); // Purpur - Barrels and enderchests 6 rows
+        for (int i = 0; i < storageSlotCount; i++) {
+            // Purpur end - Barrels and enderchests 6 rows
             ItemStack item = this.getItem(i);
             if (!item.isEmpty()) {
                 output.add(new ItemStackWithSlot(i, item));
